lib/_http_client.js

```diff
@@ -82,7 +82,8 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
-  } else if (options instanceof url.URL) {
+  } else if (options && options[searchParamsSymbol] &&
+             options[searchParamsSymbol][searchParamsSymbol]) {
```

Comment:
There's not a whole lot that can be done about that. Using a helper function as previously noted incurs a measurable performance hit.